packages/tfchain_client/src/errors.ts

```diff
@@ -27,12 +27,28 @@ class TFChainError extends Error {
   constructor(options: ITFChainError) {
     super(options.message);
     this.name = "TFChainError";
-    this.keyError = options.keyError || "GenericError";
-    options.keyError && Object.defineProperty(this, "keyError", options.keyError);
-    options.section && Object.defineProperty(this, "section", options.section);
-    options.args && Object.defineProperty(this, "args", options.args);
-    options.method && Object.defineProperty(this, "method", options.method);
-    options.docs && Object.defineProperty(this, "docs", options.docs);
+
+    if (options.keyError) {
+      Object.defineProperty(this, "keyError", { value: options.keyError, enumerable: true });
+    } else {
+      Object.defineProperty(this, "keyError", { value: "GenericError", enumerable: true });
+    }
+
+    if (options.section) {
+      Object.defineProperty(this, "section", { value: options.section, enumerable: true });
+    }
+
+    if (options.args) {
+      Object.defineProperty(this, "args", { value: options.args, enumerable: true });
+    }
+
+    if (options.method) {
+      Object.defineProperty(this, "method", { value: options.method, enumerable: true });
+    }
+
+    if (options.docs) {
+      Object.defineProperty(this, "docs", { value: options.docs, enumerable: true });
+    }
```

Comment:
Instead of the repitition, can we do sth like:

```suggestion
    const { keyError = options.keyError || "GenericError", section, args, method, docs } = options;

    Object.defineProperties(this, {
      keyError: { value: keyError, enumerable: true },
      section: { value: section, enumerable: true },
      args: { value: args, enumerable: true },
      method: { value: method, enumerable: true },
      docs: { value: docs, enumerable: true },
    });
```

Reply:
Will discard the comment as we need to define properties only if its correspondent option is defined.

Reply:
I didn't test it but maybe something like that
```ts
    if (!options.keyError) {
      options.keyError = "GenericError"
    }
    for (const key of ["keyError", "section", "args", "method", "docs"]) {
      if (options[key]) {
        Object.defineProperty(this, key, { value: options[key], enumerable: true });
      }
```

Reply:
Pretty well. I also think that defining the key dynamically would enhance the code's scalability for future use.

```ts
class TFChainError extends Error {
  constructor(options: ITFChainError) {
    super(options.message);
    this.name = "TFChainError";

    if (!options.keyError) {
      options.keyError = "GenericError";
    }

    Object.keys(options).forEach(key => {
      if (options[key as keyof ITFChainError]) {
        Object.defineProperty(this, key, {
          value: options[key as keyof ITFChainError],
          enumerable: true,
        });
      }
    });
  }
}
```